src/lib.rs

```diff
@@ -56,6 +56,20 @@ where
     Err(())
 }
 
+#[cfg(target_env = "gnu")]
+mod glibc {
+    // Attempt to compile a c program that links to strlcpy from the std
+    // library to determine whether glibc packages it.
+    pub(crate) fn has_strlcpy() -> bool {
+        let mut build = cc::Build::new();
+        build.cargo_metadata(false);
+        build.warnings_into_errors(true);
+
+        build.file("src/strlcpy.c");
+        build.try_compile("has_strlcpy").is_ok()
+    }
+}
```

Comment:
```suggestion
mod glibc {
    use std::path::Path;

    // Attempt to compile a c program that links to strlcpy from the std
    // library to determine whether glibc packages it.
    pub(crate) fn has_strlcpy() -> bool {
        let path = Path::new(env!("CARGO_MANIFEST_DIR")).join("src/strlcpy.c");

        cc::Build::new()
            .cargo_metadata(false)
            .warnings_into_errors(true)
            .file(path)
            .try_compile("has_strlcpy")
            .is_ok()
    }
}
```

Reply:
Yeah, I guess this would reduce the pollution in the out dir.